sql/server.go

```diff
@@ -116,67 +116,78 @@ func (s server) execStmt(stmt parser.Statement, params parameters, planMaker *pl
 			return result, errTransactionAborted
 		}
 	}
+
 	// Bind all the placeholder variables in the stmt to actual values.
 	if err := parser.FillArgs(stmt, params); err != nil {
 		return result, err
 	}
-	var plan planNode
-	// If there is a pending transaction.
-	if planMaker.txn != nil {
-		// Run in transaction planMaker.txn
-		var err error
-		if plan, err = planMaker.makePlan(stmt); err != nil {
-			return result, err
-		}
-	} else {
-		// No transaction. Run the command as a retryable block in an
-		// auto-transaction.
-		if err := s.db.Txn(func(txn *client.Txn) error {
-			planMaker.txn = txn
-			var err error
-			plan, err = planMaker.makePlan(stmt)
-			planMaker.txn = nil
+
+	// Create a function which both makes and executes the plan, populating
+	// result.
+	//
+	// TODO(pmattis): Should this be a separate function? Perhaps we should move
+	// some of the common code back out into execStmts and have execStmt contain
+	// only the body of this closure.
+	f := func() error {
```

Comment:
I'm not sure how to test this change. The previous code was working, though accidentally.